src/database/schemas.ts

```diff
@@ -13,132 +13,152 @@ const typeMap = {
-        `from QSYS2.SQLPACKAGE`,
-        `where PACKAGE_SCHEMA = '${schema}' ${details.filter ? `and PACKAGE_NAME like '%${details.filter}%'`: ``}`,
-        `order by PACKAGE_NAME asc`
-      ].join(` `);
-      break;
-
-    case `triggers`:
-      query = [
-        `select TRIGGER_NAME as NAME, TRIGGER_TEXT as TEXT, EVENT_OBJECT_SCHEMA as BASE_SCHEMA, EVENT_OBJECT_TABLE as BASE_OBJ `,
-        `from QSYS2.SYSTRIGGERS`,
-        `where TRIGGER_SCHEMA = '${schema}' ${details.filter ? `and TRIGGER_NAME like '%${details.filter}%'`: ``}`,
-        `order by TRIGGER_NAME asc`
-      ].join(` `);
-      break;
-
-    case `types`:
-      query = [
-        `select USER_DEFINED_TYPE_NAME as NAME, TYPE_TEXT as TEXT, SYSTEM_TYPE_NAME as SYS_NAME, SYSTEM_TYPE_SCHEMA as SYS_SCHEMA `,
-        `from QSYS2.SYSTYPES`,
-        `where USER_DEFINED_TYPE_SCHEMA = '${schema}' ${details.filter ? `and USER_DEFINED_TYPE_NAME like '%${details.filter}%'`: ``}`,
-        `order by USER_DEFINED_TYPE_NAME asc`
-      ].join(` `);
-      break;
+
+    for (const type of types) {
+      switch (type) {
+        case `schemas`:
+          selects.push([
+            `select '${type}' as OBJ_TYPE, SCHEMA_NAME as NAME, SCHEMA_TEXT as TEXT, SYSTEM_SCHEMA_NAME as SYS_NAME, '' as SYS_SCHEMA, `,
+            `  '' as SPECNAME, '' as BASE_SCHEMA, '' as BASE_OBJ`,
+            `from QSYS2.SYSSCHEMAS`,
+            details.filter ? `where SCHEMA_NAME = '${details.filter}' or SYSTEM_SCHEMA_NAME = '${details.filter}'` : ``,
+            `order by QSYS2.DELIMIT_NAME(SCHEMA_NAME) asc`
+          ].join(` `));
+            break;
+            
+      case `tables`:
+      case `views`:
+      case `aliases`:
+        selects.push([
+          `select '${type}' as OBJ_TYPE, TABLE_NAME as NAME, TABLE_TEXT as TEXT, BASE_TABLE_SCHEMA as BASE_SCHEMA, BASE_TABLE_NAME as BASE_OBJ, SYSTEM_TABLE_NAME as SYS_NAME, SYSTEM_TABLE_SCHEMA as SYS_SCHEMA, `,
+          `  '' as SPECNAME`,
+          `from QSYS2.SYSTABLES`,
+          `where TABLE_SCHEMA = '${schema}' and TABLE_TYPE in (${typeMap[type].map(item => `'${item}'`).join(`, `)}) ${details.filter ? `and TABLE_NAME like '%${details.filter}%'`: ``}`,
+          `order by TABLE_NAME asc`
+        ].join(` `));
+        break;
+  
+      case `constraints`:
+        selects.push([
+          `select '${type}' as OBJ_TYPE, CONSTRAINT_NAME as NAME, CONSTRAINT_TEXT as TEXT, TABLE_SCHEMA as BASE_SCHEMA, TABLE_NAME as BASE_OBJ, SYSTEM_TABLE_NAME as SYS_NAME, SYSTEM_TABLE_SCHEMA as SYS_SCHEMA, `,
+          `  '' as SPECNAME`,
+          `from QSYS2.SYSCST`,
+          `where CONSTRAINT_SCHEMA = '${schema}' ${details.filter ? `and CONSTRAINT_NAME like '%${details.filter}%'`: ``}`,
+          `order by CONSTRAINT_NAME asc`
+        ].join(` `));
+        break;
+  
+      case `functions`:
+        selects.push([
+          `select '${type}' as OBJ_TYPE, ROUTINE_NAME as NAME, SPECIFIC_NAME as SPECNAME, coalesce(ROUTINE_TEXT, LONG_COMMENT) as TEXT, `,
```

Comment:
Moving `, SPECIFIC_NAME as SPECNAME` to the last column position fixes the issue with specific name missing when retrieving parameters.
Same comment for the procedure query below.
General comment/rule is that with the `UNION` of all these queries, the column order within each should be the same.